### PR TITLE
Added Chapter1

### PR DESCRIPTION
**Pull request by cheezi0747**
## Description

Adds the Chapter 1 keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechstudio/chapter1/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        _______ , _______ , _______ , _______ , _______ , _______ , _______ , _______ , _______ , _______ , _______ , _______ , _______ ,           _______ , _______ ,
+        _______ , _______ , _______ ,                               _______ ,                               _______ , _______ ,           _______ , _______ , _______ ),
+};
+
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] = {ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)},
+    [1] = {ENCODER_CCW_CW(_______, _______)},
+};
```
Comment (waffle87) [inline keyboards/mechstudio/chapter1/keymaps/default/keymap.c:39]:
```suggestion
        KC_LSFT, KC_NUBS, KC_Z   , KC_X   , KC_C   , KC_V   , KC_B   , KC_N   , KC_M   , KC_COMM, KC_DOT , KC_SLSH, KC_RSFT,          KC_UP,   _______,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC ,                            KC_RALT, KC_RCTL         , KC_LEFT, KC_DOWN, KC_RGHT)
};


#if defined(ENCODER_MAP_ENABLE)
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [0] = {ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)}
};
```

File: keyboards/mechstudio/chapter1/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        _______ , _______ , _______ , _______ , _______ , _______ , _______ , _______ , _______ , _______ , _______ , _______ , _______ ,           _______ , _______ ,
+        _______ , _______ , _______ ,                               _______ ,                               _______ , _______ ,           _______ , _______ , _______ )
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] = {ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)},
+    [1] = {ENCODER_CCW_CW(_______, _______)},
+    [2] = {ENCODER_CCW_CW(_______, _______)},
+    [3] = {ENCODER_CCW_CW(_______, _______)},
```
Comment (waffle87) [inline keyboards/mechstudio/chapter1/keymaps/via/keymap.c:53]:
```suggestion
        KC_LSFT, KC_LSFT, KC_Z   , KC_X   , KC_C   , KC_V   , KC_B   , KC_N   , KC_M   , KC_COMM, KC_DOT , KC_SLSH, KC_RSFT,          KC_UP,   _______,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC ,                            KC_RALT, KC_RCTL         , KC_LEFT, KC_DOWN, KC_RGHT)
};

#if defined(ENCODER_MAP_ENABLE)
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [0] = {ENCODER_CCW_CW(KC_MS_WH_UP, KC_MS_WH_DOWN)}
```

File: keyboards/mechstudio/chapter1/rules.mk [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (waffle87) [inline keyboards/mechstudio/chapter1/rules.mk:1]:
```suggestion
# This file intentionally left blank
```

File: keyboards/mechstudio/chapter1/info.json [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+        "pid": "0x0006",
+        "device_version": "0.0.1"
+    },
+    "features": {
+        "extrakey": true,
+        "bootmagic": true,
+        "console": true,
+        "caps_word": true,
+        "mousekey": true,
+        "nkro": true
```
Comment (waffle87) [inline keyboards/mechstudio/chapter1/info.json:17]:
```suggestion
        "nkro": true,
        "encoder": true
```

File: keyboards/mechstudio/chapter1/info.json [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+    "bootmagic": {
+        "matrix": [1, 0]
+    },
+    "matrix_pins": {
+        "cols": ["B3", "F4", "F5", "C7", "C6", "B6", "B5", "B4", "D7", "D6", "D4", "D5", "D3", "D2", "D1", "D0"],
+        "rows": ["F1", "F0", "E6", "B1", "B2"]
+    },
+    "encoder": {
+        "enabled": true,
+        "rotary": [{ "pin_a": "F6", "pin_b": "F7" }]
```
Comment (waffle87) [inline keyboards/mechstudio/chapter1/info.json:28]:
```suggestion
        "rotary": [
            { "pin_a": "F6", "pin_b": "F7" }
        ]
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/mechstudio/chapter1/info.json [hunk 6]
```diff
@@ -0,0 +1,402 @@
...
+    "maintainer": "Cheezi0747",
+    "usb": {
+        "vid": "0x4D53",
+        "pid": "0x0006",
+        "device_version": "0.0.1"
+    },
+    "features": {
+        "extrakey": true,
+        "bootmagic": true,
+        "console": true,
```
[the remark this answers is not in the document]
Comment (cheezi0747) [reply keyboards/mechstudio/chapter1/info.json:14]: Nope. That is an oversight on my part. Sorry about that.

**Review by waffle87 — APPROVED** [review]

File: keyboards/mechstudio/chapter1/info.json
Comment (fauxpark) [inline keyboards/mechstudio/chapter1/info.json:1]: Please run `qmk format-json -i` on this file.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by casuanoob — APPROVED** [review]
Looks good to me.
Only thing I see is that bootmagic is apparently configured for matrix `[1, 0]` but the readme bootmagic instruction retains the default "matrix position 0, 0" boilerplate.

**Review by drashna — APPROVED** [review]

**Pull request merged**